src/runtime/runtime_unix.go

```diff
@@ -307,3 +326,164 @@ func growHeap() bool {
 	setHeapEnd(heapStart + heapSize)
 	return true
 }
+
+func init() {
+	// Set up a channel to receive signals into.
+	signalChan = make(chan uint32, 1)
+}
+
+var signalChan chan uint32
+
+// Simple boolean that's true when any signals have been registered.
+var hasSignals uint32
+
+// Mask of signals that have been received. The signal handler atomically ORs
+// signals into this value.
+var receivedSignals uint32
+
+//go:linkname signal_enable os/signal.signal_enable
+func signal_enable(s uint32) {
+	if s >= 32 {
+		// TODO: to support higher signal numbers, we need to turn
+		// receivedSignals into a uint32 array.
+		runtimePanicAt(returnAddress(0), "unsupported signal number")
+	}
+	atomic.StoreUint32(&hasSignals, 1)
+	// It's easier to implement this function in C.
+	tinygo_signal_enable(s)
+}
+
+//go:linkname signal_ignore os/signal.signal_ignore
+func signal_ignore(s uint32) {
+	if s >= 32 {
+		// TODO: to support higher signal numbers, we need to turn
+		// receivedSignals into a uint32 array.
+		runtimePanicAt(returnAddress(0), "unsupported signal number")
+	}
+	tinygo_signal_ignore(s)
+}
+
+//go:linkname signal_disable os/signal.signal_disable
+func signal_disable(s uint32) {
+	if s >= 32 {
+		// TODO: to support higher signal numbers, we need to turn
+		// receivedSignals into a uint32 array.
+		runtimePanicAt(returnAddress(0), "unsupported signal number")
+	}
+	tinygo_signal_disable(s)
+}
+
```

Comment:
We might also want to add [signal_ignored](https://cs.opensource.google/go/go/+/master:src/os/signal/signal_unix.go;drc=a9bd84e03786e8bd4bfb3780ce3283b0f3e1f3bf;bpv=0;bpt=1;l=18). This could implemented via a bitmap as well.

```suggestion
func signal_ignored(s uint32) 
```

Reply:
Do you have a code sample that requires this?
The test includes `signal.Ignore(...)` function, which doesn't result in `signal_ignored` being called.

Reply:
I had something in the back of my mind but I had a look at my build logs and could not find it, so let's leave it for now.